xarray/backends/api.py

```diff
@@ -594,6 +599,14 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
     if not paths:
         raise IOError('no files to open')
 
+    # If infer_order_from_coords=True then this is unnecessary, but quick.
+    # If infer_order_from_coords=False then this creates a flat list which is
+    # easier to iterate over, while saving the originally-supplied structure
+    combined_ids_paths, concat_dims = _infer_concat_order_from_positions\
+        (paths, concat_dims)
```

Comment:
Style note: per PEP8, prefer adding extra or moving parentheses `()` rather than explicit continuations. This could be:
```python
    combined_ids_paths, concat_dims = _infer_concat_order_from_positions(
        paths, concat_dims)
```